backend/coins/eth/account.go

```diff
@@ -156,9 +156,21 @@ func (account *Account) Initialize() error {
 	account.db = db
 	account.log.Debugf("Opened the database '%s' to persist the transactions.", dbName)
 
-	account.address = Address{
-		Address: crypto.PubkeyToAddress(*account.signingConfiguration.PublicKeys()[0].ToECDSA()),
+	addressBased := account.signingConfiguration.IsAddressBased()
+
+	if addressBased {
```

Comment:
just do `if account.signingConfiguration.IsAddressBased() {`? no need for a variable which is only used once.